src/common/fixedsizeringbuffer.h

```diff
@@ -101,25 +101,23 @@ class FixedSizeRingBuffer
      */
     void remove_if(const std::function<bool(const TYPE &)> &f)
     {
-        size_t pos = 0;
+        // filter and sort the data
+        auto tmp = decltype(_data) {};
+        auto pos = 0;
```

Comment:
`auto` will most likely make this an `int`, since the type is inferred from the `0`. It should be explicitly declared as `size_t`, I think. Otherwise, it might overflow with really really large buffers.